### ISSUE TITLE
Specify for a variety of authorisation mechanisms 

### ISSUE THREAD

**Issue by vanrein** [remotestorage/spec#39]
The RemoteStorage 01 spec describes sessions with a strict dependency on one particular authorisation mechanism, OAuth2's Implicit Grants.  Such a directed choice should either be loosened or explained, I think.  There may be solid reasons to make a deliberate choice based on the problem domain at hand -- but none I can think of.

One specific form of authorisation that is not supported (but ought to be, I think) is authorisation in the backend, away from the web sphere.  If we want RemoteStorage provided by various parties across the Internet, as well as do it ourselves and see network administrators run services for it, then we should give them a way to circumvent the security hazards and added delays of hopping from one secure website to another.

To spell it out: The OAuth model centralises computation stress on server-side TLS, it decentralises credentials, decentralises credential caching and introduces multiple parties who need to do a lot of talking and verifying each other.  It is not ideal ;-) and specifically in many situations where we would like to see RemoteStorage, it is likely that administrators don't want this overhead on top of relatively light storage services.

An obvious alternative approach would be to stick to filesystem permissions, or ACLs which may or may not be centrally managed in a directory.  This gives more reliable policy enforcement, it adds no overhead relative to file systems, and the only added responsibilities for supporting RemoteStorage is authenticated HTTP.  Depending on the local situation and personal preference, this may not even require TLS.

Imagine how simple it then becomes to serve RemoteStorage on a trusted local network: Simply layer the HTTP protocols over a local filesystem or Samba share, incorporate something as lightweight as Kerberos authentication over HTTP, and after registering it for webfinger your RemoteStorage is ready to go!  This would even be possible in an embedded/router environment, which already offer Samba-mounts for USB-sticks.  Current RemoteStorage is not nearly as likely in that sort of environment.

In short, I think the strict dependency on OAuth and zooming in on Implicit Tokens is too strict, and could easily hold back the success of the RemoteStorage concept.  Both simpler and more complex schema's are likely to be useful to someone -- and should be supported in the spec and, hopefully at some point, in the JavaScript toolkit.
